driver/views.py

```diff
@@ -33,14 +35,26 @@ def post(self, request, *args, **kwargs):
         serializer = self.get_serializer(data=request.data)
         if serializer.is_valid():
             user_data = request.data["user"]
-            user_serializer = DriverUserSerializer(data=user_data)
+            user_serializer = BaseUserSerializer(data=user_data)
             if user_serializer.is_valid():
                 user = user_serializer.save()
                 user.is_driver = True
                 user.save()
                 driver = Driver.objects.create(user=user)
-                driver.preferred_method = request.data["preferred_method"]
+                driver.license = request.data["license"]
                 driver.save()
                 return Response(serializer.data, status=status.HTTP_201_CREATED)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class DriverDetailsView(generics.RetrieveAPIView):
```

Comment:
why are u still using the  "generics.RetrieveAPIView" 
when this is based on the request.user and u anyway overrides the get method 
use generic api view insted for better flow 
and remove the id lookup